[WP-2036] Agents: add enpoint information to login

bloom1 opened the pull request:
Depends-On: https://github.com/wazo-platform/wazo-agentd-client/pull/32

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Adds endpoint to agent login, wiring it through AGI module/handler, mock/integration clients, and tests to send it to agentd.
> 
> - **Handlers/Module**
>   - `login_agent(...)` now accepts `endpoint` and forwards it to `agentd_client.agents.login_agent(..., endpoint=...)`.
>   - `modules/agent_login.agent_login` parses `endpoint` from `args[4]` and passes it to the handler.
> - **Integration helpers**
>   - `AgentdMockClient.expect_agent_login(...)` includes `endpoint` in request body.
>   - `AgidClient.agent_login(...)` signature extended to include `endpoint` and sends it over FastAGI.
> - **Tests**
>   - Unit/integration tests updated to provide/validate `endpoint` in agent login calls.
> - **Misc**
>   - Copyright years bumped to 2025.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 113461ecfae66aa7ce1f20f8acd22dd574a85375. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

bloom1 committed:
agents: add endpoint information to agent login

DrPyser commented on integration_tests/suite/test_handlers.py:
```diff
@@ -236,18 +236,21 @@ def test_agent_login(base_asset: BaseAssetLaunchingHelper):
         agent['tenant_uuid'],
         extension['context'],
         extension['exten'],
+        line['name'],
```
even though the interface is mocked, might be better to update this test with the `{chantech}/` prefix on endpoint to match real use and not be confusing.

DrPyser reviewed (commented):
only nitpicks, tested to work

DrPyser committed:
test_agent_login: prefix line name with chantech

to reflect that the actual value being expected is an asterisk endpoint identifier

DrPyser reviewed (approved):
all good, tested ok
